File: app/src/main/java/com/ss/universitiesdirectory/Fragment/NewsFragment.java
package com.ss.universitiesdirectory.Fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;
import androidx.fragment.app.Fragment;
import com.ss.universitiesdirectory.R;
import com.ss.universitiesdirectory.RSS.Downloader;

public class NewsFragment extends Fragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_news, container, false);

        Bundle bundle = this.getArguments();
        ListView listView = view.findViewById(R.id.NewsList);

        if(bundle.getString("News") != null){
            new Downloader(getActivity(), bundle.getString("News"), listView).execute();
        }

        return view;
    }
}

File: app/src/main/java/com/ss/universitiesdirectory/UI/Fragment/NewsFragment.kt
package com.ss.universitiesdirectory.UI.Fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.navArgs
import androidx.recyclerview.widget.LinearLayoutManager
import com.ss.universitiesdirectory.RSS.Downloader
import com.ss.universitiesdirectory.databinding.FragmentNewsBinding

class NewsFragment : Fragment() {

    private var _binding: FragmentNewsBinding? = null
    private val binding get() = _binding!!
    private val argument by navArgs<NewsFragmentArgs>()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        _binding = FragmentNewsBinding.inflate(inflater, container, false)

        init()
        showData()

        return binding.root
    }

    private fun init() {
        binding.newsList.layoutManager = LinearLayoutManager(requireContext())
    }

    private fun showData() {
        Downloader(argument.rssUrl, binding.newsList).execute()
    }

    override fun onDestroy() {
        super.onDestroy()
        _binding = null
    }
}